src/lamp_py/utils/filter_bank.py

```diff
@@ -1,5 +1,77 @@
+from datetime import datetime
+from typing import List, Optional
 import pyarrow.compute as pc
 import pyarrow as pa
+import polars as pl
+
+GTFS_ARCHIVE_FILTER_BANK = "s3://mbta-performance/lamp/gtfs_archive"
+# GTFS_ARCHIVE = "https://performancedata.mbta.com/lamp/gtfs_archive"
+
+
+def list_station_child_stops_from_gtfs(
+    stops: pl.DataFrame, parent_station: str, additional_filter: Optional[pl.Expr] = None
+) -> pl.DataFrame:
+    """
+    Filter gtfs stops by parent_station string, and additional filter if available
+    """
+    df_parent_station = stops.filter(pl.col("parent_station") == parent_station)
+    if additional_filter is not None:
+        df_parent_station = df_parent_station.filter(additional_filter)
+    return df_parent_station
+
+
+class LightRailFilter:
+    """
+    Data-only class for lists of filters relevant for light rail
+    """
+
+    terminal_stop_ids = list(map(str, [70106, 70160, 70161, 70238, 70276, 70503, 70504, 70511, 70512]))
+
+
+class HeavyRailFilter:
+    """
+    Data-only class for lists of filters relevant for heavy rail
+    """
+
+    def __init__(self) -> None:
+
+        self._terminal_stop_ids_numeric = list(map(str, [70001, 70036, 70038, 70059, 70061, 70094, 70105]))
+        self._terminal_stop_place_names = [
+            "place-forhl",
+            "place-ogmnl",
+            "place-bomnl",
+            "place-wondl",
+            "place-alfcl",
+            "place-asmnl",
+            "place-brntn",
+        ]
+        self.terminal_stop_ids_list: list = []
+
+    def terminal_stop_ids(self) -> List:
+        """
+        populate on first call - otherwise return populated
+        """
+        if not self.terminal_stop_ids_list:
+            self.current_terminal_stop_ids()
+        return self.terminal_stop_ids_list
+
+    def current_terminal_stop_ids(self) -> None:
+        """
+        When called, dynamically populates terminal_stop_ids based on the
+        _terminal_stop_place_names. This has to live outside of the constructor
+        because we want to be able to patch the GTFS_ARCHIVE_FILTER_BANK for
+        unit tests (s3 path vs public performancedata path) which is not possible
+        when terminal_stop_ids is populated on class construction.
+        """
+        heavy_rail_filter = pl.col("vehicle_type") == 1
+
+        service_date = datetime.now()
+        stops = pl.read_parquet(f"{GTFS_ARCHIVE_FILTER_BANK}/{service_date.year}/stops.parquet")
```

Comment:
Reading this file should also be filtered by a specific service date. While the `stops` file is relatively stable. Not filtering by service date may eventually result in unexpected results. 

I think there is also an open ticket on the parquet GTFS Archive files where when a new year rolls over, a new year parquet file is not automatically created, so there is the potential for this call to fail during the first days of a new year.

Reply:
the `gtfs_from_parquet` function in `gtfs_utils.py` accounts for all of this.